modules/packages/LocalAtomics.chpl

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module LocalAtomics {
+
+  if CHPL_TARGET_ARCH != "x86_64" {
+    compilerWarning("Program compiled for ", CHPL_TARGET_ARCH, " but target should be x86_64");
```

Comment:
The compile-time check here is good, but the message could be better.  The current phrasing implies that the user just needs to change their target.  That's not so, and indeed may not be practical: they may not have a choice as to which machine they're using.  Instead it should say something like:
```
compilerWarning("The LocaleAtomics package module cannot support CHPL_TARGET_ARCH=", CHPL_TARGET_ARCH, ", only x86_64.");
```
That tells them the limitation is in the package module, while still letting them know that if they can change targets that will let them proceed.

Reply:
Thanks for the suggestion @gbtitus ! I made the changes you requested :)